src/TestFramework/MSTest.Core/Attributes/DynamicDataAttribute.cs

```diff
@@ -131,7 +141,34 @@ public IEnumerable<object[]> GetData(MethodInfo methodInfo)
         /// <inheritdoc />
         public string GetDisplayName(MethodInfo methodInfo, object[] data)
         {
-            if (data != null)
+            if (this.DynamicDataDisplayName != null)
+            {
+                var dynamicDisplayNameDeclaringType = this.DynamicDataDisplayNameDeclaringType ?? methodInfo.DeclaringType;
+
+                var method = dynamicDisplayNameDeclaringType.GetTypeInfo().GetDeclaredMethod(this.DynamicDataDisplayName);
+                if (method == null)
+                {
+                    throw new ArgumentNullException(string.Format("{0} {1}", DynamicDataSourceType.Method, this.DynamicDataDisplayName));
+                }
+
+                var parameters = method.GetParameters();
+                if (parameters.Length != 2 ||
+                    parameters[0].ParameterType != typeof(MethodInfo) ||
+                    parameters[1].ParameterType != typeof(object[]) ||
+                    method.ReturnType != typeof(string) ||
+                    !method.IsStatic)
+                {
+                    throw new ArgumentNullException(
+                        string.Format(
+                            FrameworkMessages.DynamicDataDisplayName,
+                            this.DynamicDataDisplayName,
```

Comment:
Method signature should also have "public" access modifier specified. Can you please add that as well??

Reply:
You will have to update xlf files as well if you modify  FrameworkMessages.DynamicDataDisplayName string.

Reply:
Done. Interestingly it did work with a private static method even though the documentation for `GetDeclaredMethod` states that "Returns an object that represents the specified **public** method declared by the current type." I have updated the exception message and added a test for public accessibility.

Reply:
Super 👍